firtool-resolver/src/Main.scala

```diff
@@ -69,22 +69,29 @@ object Resolve {
     new File(path).getAbsolutePath()
   }
 
-  private def checkPath(logger: Logger): Either[String, FirtoolBinary] = {
+  // Nested Either is weird but represents unrecoverable vs. recoverable errors
+  // Left(...) - unrecoverable error
+  // Right(Left(...)) - recoverable error
+  // Right(Right(...)) - success
```

Comment:
This would likely be clearer as custom sum types that represent exactly this (unrecoverable error, recoverable error, and success) as opposed to using nested either. Not a huge problem if not.

Reply:
I agree that's better, it's a decent chunk of work though. Depending on how much more development we give to this project we can consider it. If we do end up generalizing this to support espresso and verilator, we should definitely make that improvement.